lib/mcb/grant_access_wizard.rb

```diff
@@ -19,16 +20,20 @@ def fetch_organisation
       @organisation = @provider.organisations.first # a provider should only ever be associated with one organisation
     end
 
-    def ask_email
-      @email = @cli.ask("Email address of user?  ").strip.downcase
-    end
-
     def find_or_init_user
-      @user = User.find_or_initialize_by(email: @email) do |u|
-        puts "#{@email} appears to be a new user"
-        u.first_name = @cli.ask("First name?  ").strip
-        u.last_name = @cli.ask("Last name?  ").strip
+      @user = MCB.find_user_by_identifier @id_or_email_or_sign_in_id # is it okay to access MCB from here? I suspect not
+      return @user if @user != nil
+
+      unless @id_or_email_or_sign_in_id.include? '@'
+        puts "#{@id_or_email_or_sign_in_id} not found. Specify an email address if you wish to create a user"
+        return nil
       end
+
+      @user = User.new(email: @id_or_email_or_sign_in_id)
+      puts "#{@email} appears to be a new user"
+      @user.first_name = @cli.ask("First name?  ").strip
+      @user.last_name = @cli.ask("Last name?  ").strip
+      @user
```

Comment:
What about putting this into `lib/mcb` somewhere so that it can be used by other commands?

Reply:
can we come back to this? I've spent ages on this already and I still need to do the actual revoke command